bad-cert.js

```diff
@@ -0,0 +1,23 @@
+var tls = require('tls');
+var https = require('https');
+
+var socket1 = tls.connect({
+    port: 1337,
+    host: 'https://example1.com',
+    rejectUnauthorized: true,
+    checkServerIdentity: function(servername, peer) {  
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- d04f008c70596e2d064ca1db78bcd52c -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Improper Certificate Validation
_Medium CWE-295_
The application does not verify if the server identity matches the provided certificate.

Improper certificate verification may allow an attacker to spoof a trusted entity by using a man-in-the-middle attack.

### How to fix
Potential mitigations include the following:
  1. Certificates should be carefully managed and checked to assure that data are encrypted with the intended owner's public key.
  2. If certificate pinning is being used, ensure that all relevant properties of the certificate are fully validated before the certificate is pinned, including the hostname.